packages/lib-classifier/src/components/Classifier/components/Banners/Banners.js

```diff
@@ -11,14 +11,32 @@ function useStores(stores) {
   const { classifierStore } = stores ?? React.useContext(MobXProviderContext)
   return {
     subject: classifierStore.subjects.active,
+    subjects: classifierStore.subjects,
     workflow: classifierStore.workflows.active
   }
 }
 
+const environment = process.env.APP_ENV
+
 function Banners({ stores }) {
-  const { subject, workflow } = useStores(stores)
+  const { subject, subjects, workflow } = useStores(stores)
   const subjectNumber = subject?.priority ?? -1
-  if (workflow?.grouped && subjectNumber !== -1) {
+  const hasIndexedSubjects = workflow?.hasIndexedSubjects
+  const hasGroupedOrderedSubjects = workflow?.grouped && workflow?.prioritized
+  if (!subject) {
+    return null
+  }
```

Comment:
We want the banners to wait on the subject data, otherwise we'll get `undefined` printed in the banner for the subject priority number. Or we'll get an ugly flash of a green banner, while `subject.already_seen` and `subject.retired` are undefined, which then changes to red for finished subjects. You might be able to check this on master, at the moment, for finished subjects in HMS NHS. I'm not sure.

I'm checking for the existence of a subject model because I don't think individual subjects have a loading state. I could use `store.subjects.loadingState` though.

Reply:
The final loading state of the subject is the viewer state onload of the media. If this fails, then it doesn't make sense to show the banner.

Reply:
True. The task area is also enabled before the subject media has loaded too. 

I think that’s linked to video not having a load event, but I’m not sure. So we’d have to be careful about not breaking the banners for data or video subjects.

Reply:
The `SingleVideoViewer` and all the JSON subject viewers have `onLoad` events defined and used.

Reply:
Ok. Then there’s a bug where you can start to classify before those events fire. I came across that while writing the tests for the task area. Tasks should be disabled until the subject has finished loading, but aren’t.

Reply:
On second thoughts, I've taken this out. The `Banners` component is already set up to render an empty string when `subject` is undefined, so I've added a test for that instead. On master, there's no test for what happens in that case.